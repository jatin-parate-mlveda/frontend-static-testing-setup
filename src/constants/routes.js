export const home = '/home';
export const blog = '/blog-detail';
