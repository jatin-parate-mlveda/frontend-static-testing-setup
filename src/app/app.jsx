import { lazy, Suspense } from 'react';
import { Route, Switch, NavLink, Redirect } from 'react-router-dom';
import { home, blog } from 'constants/routes';
import { ErrorBoundary } from 'shared/error-boundary';

// import logo from '../assets/images/logo.svg';
import classes from './app.module.scss';

const BlogDetail = lazy(() =>
  import(/* webpackPrefetch: true */ '../pages/blog-detail/blog-detail').then(
    module => ({
      default: module.BlogDetail,
    }),
  ),
);

const HomePage = lazy(() =>
  import(/* webpackPrefetch: true */ '../pages/home-page/home-page').then(
    module => ({
      default: module.HomePage,
    }),
  ),
);

export function App() {
  return (
    <ErrorBoundary>
      <div className={classes.app}>
        <header className={classes.app__header}>
          <NavLink className={classes.app__link} to={home}>
            Home
          </NavLink>
          <NavLink className={classes.app__link} to={blog}>
            Blog
          </NavLink>
        </header>
        <Switch>
          <Route
            path={blog}
            render={() => (
              <Suspense fallback={<h1>Loading...</h1>}>
                <BlogDetail />
              </Suspense>
            )}
          />
          <Route
            path={home}
            render={() => (
              <Suspense fallback={<h1>Loading...</h1>}>
                <HomePage />
              </Suspense>
            )}
          />
          <Route path='/' exact>
            <Redirect to={home} />
          </Route>
        </Switch>
      </div>
    </ErrorBoundary>
  );
}
