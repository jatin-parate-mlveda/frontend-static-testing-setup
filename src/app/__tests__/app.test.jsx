import { render } from '@testing-library/react';
import { blog } from 'constants/routes';
import { MemoryRouter } from 'react-router-dom';
import { App } from '../app';
import classes from '../app.module.scss';
import { waitFor2Sec } from '../../../test/utils/wait-for-2sec';

test('It renders loading state successfully', async () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );
  const eles = container.getElementsByClassName(classes.app);

  expect(eles).toHaveLength(1);

  expect(container.outerHTML).toMatchSnapshot();
});

test('It renders successfully', async () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );

  await waitFor2Sec();

  const eles = container.getElementsByClassName(classes.app);

  expect(eles).toHaveLength(1);

  expect(container.outerHTML).toMatchSnapshot();
});

test('It renders blogs page loading state', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={[{ pathname: blog }]}>
      <App />
    </MemoryRouter>,
  );

  expect(container.outerHTML).toMatchSnapshot();
});

test('It renders blogs page successfully', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={[{ pathname: blog }]}>
      <App />
    </MemoryRouter>,
  );

  await waitFor2Sec();

  const eles = container.getElementsByClassName(classes.app);

  expect(eles).toHaveLength(1);

  expect(container.outerHTML).toMatchSnapshot();
});
