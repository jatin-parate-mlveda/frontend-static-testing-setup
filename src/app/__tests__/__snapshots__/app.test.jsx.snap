// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders blogs page loading state 1`] = `"<div><div class=\\"app\\"><header class=\\"app__header\\"><a class=\\"app__link\\" href=\\"/home\\">Home</a><a aria-current=\\"page\\" class=\\"app__link active\\" href=\\"/blog-detail\\">Blog</a></header><h1>Loading...</h1></div></div>"`;

exports[`It renders blogs page successfully 1`] = `"<div><div class=\\"app\\"><header class=\\"app__header\\"><a class=\\"app__link\\" href=\\"/home\\">Home</a><a aria-current=\\"page\\" class=\\"app__link active\\" href=\\"/blog-detail\\">Blog</a></header><h1>Blog Detail</h1></div></div>"`;

exports[`It renders loading state successfully 1`] = `"<div><div class=\\"app\\"><header class=\\"app__header\\"><a class=\\"app__link active\\" href=\\"/home\\" aria-current=\\"page\\">Home</a><a class=\\"app__link\\" href=\\"/blog-detail\\">Blog</a></header><h1>Loading...</h1></div></div>"`;

exports[`It renders successfully 1`] = `"<div><div class=\\"app\\"><header class=\\"app__header\\"><a class=\\"app__link active\\" href=\\"/home\\" aria-current=\\"page\\">Home</a><a class=\\"app__link\\" href=\\"/blog-detail\\">Blog</a></header><div class=\\"container\\"><form><div><label for=\\"title\\">Title<input name=\\"title\\" id=\\"title\\" value=\\"\\"></label></div><div><label for=\\"desc\\">Description<input name=\\"desc\\" id=\\"desc\\" value=\\"\\"></label></div><div><button type=\\"submit\\" class=\\" button\\">Submit</button></div></form><div><ul><li>Blog 1</li><li>Blog 2</li><li>Blog 3</li><li>Blog 4</li></ul></div></div></div></div>"`;
